### ISSUE TITLE
External libraries problem (MinGW + GLFW)

### ISSUE THREAD

**Issue by BecauseReasons** [microsoft/vscode-cpptools#3037]
Type: Debugger

Hi.

I'm trying to compile this c++ code:

```
#include <iostream>
#include <GLFW/glfw3.h>

int main()
{
    if(!glfwInit())
        return -1;

    system("pause");

    glfwTerminate();

    return 0;
}
```

**Enviro Configuration**
- OS: Windows 10 Enterprise 64-bit
- VS Code Version: 1.30.2
- C/C++ Extension Version: 0.20.1
- MingGW Version: 8.2.0
- GLFW Version: 3.2.1

**c_cpp_properties.json**
```
{
    "configurations": [
        {
            "name": "MinGW_x64",
            "intelliSenseMode": "msvc-x64",
            "compilerPath": "D:/Projects/Libraries/MinGW/bin/g++.exe",

            "windowsSdkVersion": "10.0.17763.0",
            "cStandard": "c11",
            "cppStandard": "c++17",

            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE"
            ],

            "includePath": [
                "${workspaceFolder}/**",
                "D:/Projects/Libraries/MinGW/lib/gcc/mingw32/8.2.0/include/c++",
                "D:/Projects/Libraries/glfw-3.2.1/include"
            ],

            "browse": {
                "path": [
                    "${workspaceFolder}/**",
                    "D:/Projects/Libraries/MinGW/lib/gcc/mingw32/8.2.0/include/c++",
                    "D:/Projects/Libraries/glfw-3.2.1/include"
                ],
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": ""
            }
        }
    ],
    "version": 4
}
```
**launch.json**
```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "WIN DEV",
            "type": "cppdbg",
            "request": "launch",
            "program": "${workspaceFolder}/out.exe",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": true,
            
            "MIMode": "gdb",
            "miDebuggerPath": "D:/Projects/Libraries/MinGW/bin/gdb.exe",
            "preLaunchTask": "build_dev",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ]
        }
    ]
}

```

**task.json**
```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "build_dev",
            "type": "shell",
            "command": "g++",      
            "args": [
                "-g",
                "main.cpp",
                "-ID:/Projects/Libraries/glfw-3.2.1/include",
                "-LD:/Projects/Libraries/glfw-3.2.1/lib-mingw-w64",
                "-o",
                "out.exe",
                "-lglfw3",
                "-lglfw3dll",
                "-lopengl32",
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            }
        }
    ]
}
```

Trying to compile gives me this output:
![vscode_output](https://user-images.githubusercontent.com/32194490/51183880-e672e900-18d2-11e9-990d-80efc83d94b0.PNG)

I've already tried diffrent combiantions of commands / language standards / file loctions and every time 
I'm getting the same result. 

Unfortunatelly solution presented here -> [https://github.com/Microsoft/vscode-cpptools/issues/2026](https://github.com/Microsoft/vscode-cpptools/issues/2026) did't work for me :(

**Comment by pieandcakes** [comment]
Does it work when you command line compile with the same arguments from a normal command window?

**Comment by BecauseReasons** [comment]
Well ... it does not :( I'm getting the same results.
![cmd_output](https://user-images.githubusercontent.com/32194490/51240393-e2e76c80-197b-11e9-8e9e-7cba0ff005de.PNG)

**Comment by pieandcakes** [comment]
@BecauseReasons That sounds like a tool set issue then and not an extension issue. 

I've not used the `glfw` library before so i can't speak to what needs to happen in order to include it. 

That being said, can you point to where the dlls that contain `gfw3` and `gfw3dll` live?

**Comment by BecauseReasons** [comment]
**glfw3** and **glfw3dll** are contained in: D:\Projects\Libraries\glfw-3.2.1\lib-mingw-w64

**Comment by pieandcakes** [comment]
@BecauseReasons You will need to ask for help with MinGW on how to compile and resolve that error. It looks correct to me. Since it won't compile on the command line, it seems you are missing something but like I said in the previous comment, I haven't used that before. 

A quick internet search returns [this StackOverflow](https://stackoverflow.com/questions/18280427/mingw-cant-find-glfw-libraries) post that might be of use to you.

**Comment by BecauseReasons** [comment]
I've managed to fix the issue. The code compiles using command line. Now I'll have to make it work from VSCode :D

For the record. The issue was my MinGW library version. I've downgraded it to 8.1.0 x84-x64.
Also I had some issues with drivers linking so I've added -lgdi32 option to the linker (according to this StackOverflow thread: [https://stackoverflow.com/questions/22008845/glfw-mingw-link-error]())

All in all thank you for your help. Your last response helped me to find the issue :)

Best Wishes.
